### ISSUE TITLE
Order of licenses in drop-down menu

### ISSUE THREAD

**Issue by ariel11** [clearlydefined/website#299]
With so many SPDX license options, it may be helpful to order them with the OSI-approved license options first (or in order of popularity).  For example, the OSI-approved BSD-3 clause template is listed *last* in the drop-down menu for the BSD-3 clause options.  Same with MIT, I think.  You could also add a link next to each one that would take you to the full SPDX license text for easier reference.

**Comment by jeffmcaffer** [comment]
Great idea @ariel11 . Would it make sense to essentially have two sections to the drop down, the first is the 5 most popular, then a separator, and the full list of all licenses?

Sidenote, as it is, the list comes directly from https://github.com/shinnn/spdx-license-ids/blob/master/index.json. it seems that that is largely in alphabetical order but there is a bug for MIT. It should be first in the MIT* family but is not.

**Comment by ariel11** [comment]
@jeffmcaffer - That is a good question, as I bet some of these are essentially never used.  At the same time, there might be less error if a person could see all the options together vs. having a separator.  If someone just saw GPLv2 because 'GPLv2 or later' was down the list, that may lead to errors....  For me, if the OSI Approved ones could be in bold and/or listed first, that would be helpful.

**Comment by fossygirl** [comment]
@jeffmcaffer I've opened https://github.com/spdx/license-list-data/issues/36 on SPDX.

**Comment by fossygirl** [comment]
@jeffmcaffer @storrisi I believe this can be updated/implemented now that the corresponding library has been updated. Can we take a few minutes to do this and close this out this week?

**Comment by storrisi** [comment]
@jeffmcaffer @fossygirl me and @AlexWebYourmind had checked the update done in https://github.com/clearlydefined/website/pull/379

Actually, this change alone wont fix the problem, and I'll explain why below.
In the website, we are using https://github.com/shinnn/spdx-license-ids , which is basically an exported list of ids of the official SPDX license list object file.

As mentioned in https://github.com/spdx/license-list-data/issues/36 , The SPDX license JSON is not sorted, so they preferred to change the NuGet tools in order to generate a sorted list, see here https://github.com/NuGet/NuGet.Client/pull/2504

Now, the SPDX list is updated with the new sorting as you can see [here](https://github.com/spdx/license-list-data/blob/master/json/licenses.json), but actually the spdx-license-ids library is not.

Given that [license-list-data repo](https://github.com/spdx/license-list-data) is not an installable library, in order to be up-to-date with any update of the list, we can do something manually, cloning the repo and use the json file (properly flattened) in the website  LicensePicker code, and parse it in order to retrieve only the necessary ids.
The problem is that this activity should be done manually for each update of the spdx license data.

What do you think? Shall we go with the above?

**Comment by jeffmcaffer** [comment]
Sigh. Let's just add a `sort` call in the UI. Hopefully we just sort once when we load the list.

**Comment by fossygirl** [comment]
If this ends up being complex/time-consuming, please feel free to de-prioritize and work on other things. I thought this would be a quick win and we could close out without too much work, but if it's a big deal please feel free to table it.

**Comment by jeffmcaffer** [comment]
nah, we should fix. sorting is easy and it is a real pain in the but that MIT does not sort first when you type M

**Comment by storrisi** [comment]
@jeffmcaffer if i'm not wrong, we should sort the list both for names and OSI approvement.
Given that the list that we are a using is only a flat array of strings (which are representing the ids), a simple sort can't be enough to solve the problem.

**Comment by jeffmcaffer** [comment]
not sure we care about OSI per se. ClearlyDefined is about capturing what is there etc not picking or judging. Users can understand an alphabetically sorted list. A list where the order is determined by OSI-approval will likely not be clear to people. Suggest that we try the simple sort and see if we get feedback to the contrary.

**Comment by storrisi** [comment]
@jeffmcaffer i see.
I've addressed the sorting of values into #380
